packages/chat-stateful-client/src/StatefulChatClient.ts

```diff
@@ -141,6 +146,20 @@ export const createStatefulChatClient = (
   );
 };
 
+/**
+ * A function to modify the state of the StatefulChatClient.
+ *
+ * Provided as a callback to the {@Link StatefulChatClient.modifyState} method.
+ *
+ * The function must modify the provided state in place as much as possible.
+ * Making large modifications can lead to bad performance by causing spurious rerendering of the UI.
+ *
+ * Modifier function must return true if and only if some modification was made to the state.
```

Comment:
What's the benefit of returning true/false? Can't the modifier just return the unchanged state instead and we add a ref equality check before setting it?

Reply:
This is related to the other discussion about modifier via lambda vs return value. So let's discuss there.

The way `modifyState` works, the modifier is a lambda that changes state in-place. So we don't get a new object as a return value.

Reply:
Let me ask another way:
Why can't the Context's `setState` function not do a reference equality check internally instead of relying on a Contoso provided bool flag?
The design with the bool return value leaves the pitfall open that Contoso's writes a modifier that actually modifies the state but returns false, or the other way around. I have trouble seeing the value in the bool return value.

Reply:
+100

Updated the existing `batch()` API to do this. (In a separate PR), might be worth following this emit-if-changed pattern internally as well. I foudn some spurios events being emitted in calling in my recent unittest refactor.